com.unity.render-pipelines.universal/Runtime/LightCookieManager.cs

```diff
@@ -115,6 +120,97 @@ public void Resize(int size)
             }
         }
 
+        private struct ShaderBitArray
+        {
+            const int kBitsPerElement = 32;
+            const int kElementShift = 5;
+            const int kElementMask = (1 << kElementShift) - 1;
+
+            private float[] m_data;
+
+            public int elemLength => m_data == null ? 0 : m_data.Length;
+            public int bitCapacity => elemLength * kBitsPerElement;
+            public float[] data => m_data;
+
+            public void Resize(int bitCount)
+            {
+                if (bitCapacity > bitCount)
+                    return;
+
+                int newElemCount = ((bitCount + (kBitsPerElement - 1)) / kBitsPerElement);
+                if (newElemCount == m_data?.Length)
+                    return;
+
+                var newData = new float[newElemCount];
+                if (m_data != null)
+                {
+                    for (int i = 0; i < m_data.Length; i++)
+                        newData[i] = m_data[i];
```

Comment:
Could also use [UnsafeUtility.MemCpy](https://docs.unity3d.com/ScriptReference/Unity.Collections.LowLevel.Unsafe.UnsafeUtility.MemCpy.html) iirc

Reply:
Thanks. I'll see if I can use those instead.

Reply:
It becomes a bit of hassle to do unsafe block + 2 fixed block pointers in order to use this. It's not that many elements usually (N/32), so I'll leave it as a simple for loop.